.github/actions/branch-lock/src/main.ts

```diff
@@ -0,0 +1,104 @@
+import * as core from '@actions/core'
+import {actionInputs} from 'github-actions-utils'
+import {GraphQLClient, gql} from 'graphql-request'
```

Comment:
Oktokit already have a library for GrapQL https://github.com/octokit/octokit.js#graphql-api-queries